Enhancement: Improvements to iohyve list and info

EpiJunkie opened the pull request:
This pull request contains three commits to improve the visibility of multiple pool installations. See each commit for full details.

2f2b3c7:
`iohyve info` was rewritten to include the description when spaces are present. Each pool gets it's own table of output but everything is aligned as a single block of output for readability. The `-d` flag was removed.

0548ceb
`iohyve list` now contains the pool name in the output. This is to increase visibility for multi pool installations.

1f44663
This supports the documentation (man page, text, and `iohyve help`) to reflect changes from 2f2b3c7.

EpiJunkie committed:
Enhancement: Rewrite of iohyve info

This is a partial rewrite of the `info` sub function. It now displays a per pool table of all
the variables from before and also displays the full description even if spaces are used. A
tacky backtick \` is used to create visual separation of the tables because otherwise
`column` clobbers the white space when a \\n was used instead. Removes the `-d` flag functionality.

To be honest, there would of been one table with a `$pool` field but after 2 hours of attempts,
I decided that multiple tables sounded better. This is probably possible but the problem I kept
running into was `awk` was not properly displaying the `$pool` string after column `$6`.

Here is an explanation of each pipe section for the `iohyve info` magic. The `*` in the command section
on the right indicate the remaining command to the pipe (`|`) symbol or end of line.

Excerpt:
````
(printf 'Name\tSize\tRAM\tCPU\tOS\tLoader\tDescription\n';
	zfs list -H -o name,volsize,iohyve:ram,iohyve:cpu,iohyve:os,iohyve:loader,iohyve:description | \
	grep $pool/iohyve | grep -Ev "ISO|Firmware" | cut -d '/' -f3- | sed 1d) | \
	awk '{print $1,"^",$2,"^",$3,"^",$4,"^",$5,"^",$6,"^",$1=$2=$3=$4=$5=$6="",$0 }'
````
Explanation:
````
`(` - Tells the Bourne shell to lump this together all the way to the next )
`printf *` - Print headers inline so when it reaches `column` it get aligned correctly.
`zfs list *` - Print parser friendly (`-H`) output with the properties (`-o`) with the following names
`grep $pool/iohyve` - Print lines containing active guests, disks, ISO, and Firmware datasets
`grep -Ev "ISO|Firmware"` - Print an inverse (`-v`) matches only not containing "ISO" or "Firmware"
`cut -d '/' -f3-` - Prints only the guest name and disk names
`sed 1d` - Removes first line which is the root dataset $pool/iohyve
`)` - Tells the Bourne shell to stop lumping from the openning parenthesis.
`awk *` - Print the first 6 columns with a unique charater (`^`) in between, then set the value of
         those columns to nothing, then print the remaining columns (description with spaces).
`)` - Second closing parenthesis to align multiple pool tables together. Also the reason
       for the backticks used as visual separation.
`column -ts^'` - Prints text in human friendly format using tabs (`-t`) using a specified (`s`)
                   delimiting character (`^`) instead of using the default of whitespace as the
                   delimiter.
````

Results:
````
root@bhost:~ # iohyve info
\`
iohyve guests on apool
Name                         Size    RAM       CPU    OS           Loader               Description
guestsystem                  -       256M      1      default      bhyveload            testing multi-disk
guestsystem/disk0            32G     256M      1      default      bhyveload            C Drive
guestsystem/disk1            100G    256M      1      default      bhyveload            E Drive
\`
iohyve guests on zpool
Name                         Size    RAM       CPU    OS           Loader               Description
debian                       -       4G        2      debian       grub-bhyve           Ubiquiti mFi Controller Appliance
debian/disk0                 200G    4G        2      debian       grub-bhyve           Ubiquiti mFi Controller Appliance
centos                       -       512M      1      centos7      grub-bhyve           Tue Apr 5 16:02:36 MDT 2016
centos/disk0                 10G     512M      1      centos7      grub-bhyve           Tue Apr 5 16:02:36 MDT 2016
````

EpiJunkie committed:
Fix: Updating docs to reflect change from 2f2b3c7

Updates docs to reflect the removal of `iohyve info -d` flag and also to indicate a per
pool table is displayed.

EpiJunkie committed:
ENC: `iohyve list` now displays the pool in which the guest belongs.

Results:
````
root@bhost:~ # iohyve list
Guest         VMM?  Running  rcboot?  Pool       Description
guestsystem   NO    NO       NO       apool      testing multi-disk
debian        YES   YES      NO       zpool      Ubiquiti mFi Controller Appliance
guestsystem1  NO    NO       NO       zpool      Tue Apr  5 16:02:36 MDT 2016
````

Basically duplicate names on different pools are bad, so don't do
drugs kids mmmkay. And here is why:

This commit will reveal a _insert-word_ with `iohyve`. If there are
two guests of the same name on different pools, both guests will display
`"YES"` for `VMM` and `Running` when only one is on. This is really
not a concern as fundentmentally `iohyve` would need to make many
changes for guests of the same name on different pools to work.
Currently `iohyve` does not include the pool name in the `VMM` instance,
nor does it `pgrep` for the pool when looking for a process. As a
matter of throughoutness, only the guest on the alphabetically first
pool is allowed to start. Two guests with the same name would only
occur from a `zfs send | recv` or an `iohyve rename` in which the
guest was on the alphabetically second (or more) pool (currently
there is not a check to see if the rename is itself (ie. `iohyve
rename bguest bguest`.) Quiet a few small changes can be done to
negate an accidential occurance of duplicate names with `iohyve`.

"The work continues." ~ Qyburn

pr1ntf commented:
Alright, this is a tricky one. 

I already had some list cleaning I wanted to do (briefly mentioned [here](https://github.com/pr1ntf/iohyve/wiki/General-Path-to-1.0#09-goals-submit-to-ports)).
Part of this was completely removing `description` from the `info` function and adding `tap` and `con` instead.
I would like to keep the `list` function simple, with basically what it has now, so maybe adding `pool` to the `info` function would be better. 

Last but not least, I wanted to add a `--less` flag so you could scroll through a large list that takes up too much of the terminal. 

If you do not want to make these changes, no problem, I'll add these specific points to the roadmap above. Especially the `pool` option. 

I'm open to discussion.

pr1ntf commented:
Also, I'd like to mention we generate the `iohyve.8` man page with a one-liner utilizing `txt2man`:

```
cat iohyve.8.txt | txt2man -t iohyve -s 8 -v "FreeBSD System Manager's Manual" > iohyve.8
```

EpiJunkie committed:
Updated what list does.

EpiJunkie committed:
Fixed typo in isolist definition

EpiJunkie committed:
Properly rebuilt man page file after previous commits

EpiJunkie commented:
Thank you for the heads up on the rebuilding the `man` page file. I noticed the header about the same time you posted and was trying to figure it out when I noticed your message here. Thank you.

Added commits to update a couple definitions and then a properly rebuilt `man` page file.

pr1ntf commented:
Did you happen to see my previous notes before the man page comment?

EpiJunkie commented:
I missed that post actually, thanks.

First thing I _have to_ ask, is do you really have that many guests that `--less` is necessary? Can you ballpark a number?

What if I kept the output format I have created so far but put it under `-v` for verbose and then made something more to your liking under the normal command? Also maybe use `-l` instead of `--less`.

pr1ntf commented:
```
tparis@DeltaFlyer:~/iohyve % iohyve list | wc -l
      46
tparis@DeltaFlyer:~/iohyve % 
```

They all don't run at the same time, only about ~8ish at a time, but I have different VMs for different projects and ideas. Some are clones of templates, too.  I usually just manually run `iohyve list | less` or `iohyve info | less`.  You should see my `snaplist` output. 

Let's keep the `list` function untouched, but remove `description` from the `info` function. 

Maybe info without options will output `size`, `cpu`, `ram`, and `pool`.  
Info with the `-v` flag can add `os`, `loader`, `tap`, and `con`.

EDIT: Didn't mean to close, fingers got crossed.

pr1ntf commented:
Be sure to read the previous post on GitHub, not on email, so you can see my edits before I accidentally closed this PR.

EpiJunkie committed:
Merge from upstream - 0.7.4 Stop Being a Troll Edition

EpiJunkie commented:
Wow, that's quiet a few guests.

I will work on these changes this weekend.

EpiJunkie committed:
Reverting 0548ceb: Removed $pool from iohyve list output

allenmchan commented:
Is it possible to list IPs of the VMs when using the iohyve list command?
I have all of the VMs using DHCP. It is a pain to console into each VM to get the IP

EpiJunkie commented:
Not within `iohyve`, especially when using DHCP. Maybe add static DHCP mapping to your DHCP server and then add the IP in the `description` property?

allenmchan commented:
Ok thanks for the workaround

EpiJunkie committed:
Enh: Rewrite of info() to use flags for verbosity

Complete rewrite of iohyve info to use flags to
change the verbosity level of the output. Uses
flags -p -v -s -d and -l which must be used as one
series of characters after the "-". So "-dsl", "-v"
"-dvpls" all work, "-d -v -s -l -p" will not work.
Order of flags is not important.

 -p is for pool

 -v is for verbose which adds the fields: OS,
    Loader, Tap, and Con.

 -s is for status which add the fields: VMM,
    Running, and rcboot.

 -d is for description

 -l is for less and pipes the final output to less

The way this is written, extra characters are
ignored.

EpiJunkie committed:
Doc: Updated documentation for c4487868 iohyve info rewrite

EpiJunkie committed:
Enc: Use -l with iohyve list to pipe to less

EpiJunkie committed:
Doc: Updated to reflect -l flag for list. See 1bdea39.

EpiJunkie committed:
Doc: Added what each flag does for iohyve info

EpiJunkie committed:
Enh: Reworked output for list() to use flags for verbosity

Reworked iohyve list to use flags. Removed description
from default output. With the use of flags the
description is diplayed. Flags -v -d and -l have been
added and are described below. The flags must be used
as a series of characters after the "-". So "-lvd",
"-v", "-dv" are correct syntax, however "-d -v -l"
is not. The order of the flags is not important.

-v is for verbose which adds the fields: Tap and Con

-d is for description

-l is for less and pipes the final output to less

The way this is written, extra characters are
ignored.

EpiJunkie committed:
Doc: Added information on flags for changes to list

EpiJunkie commented:
Okay I made some changes. I hope they are to your liking.

There is more code in `list()` _but_ there are flags to dial up the verbosity. The description field is removed by default as requested above. The tap and con fields are turned on by `-v`, the description field is turned back on by `-d`flag, and using the `-l` flag pipes the output to `less`. 

I also reworked `info()` as well, the code is very similar to `list()`. There is a lot more code due to the extra fields but I am very happy with the flexibility of it. The old code was so concise. This sub-function follows the same pattern as `list` where the verbosity can be dialed up using flags. The default output is to list guest name, size, CPU, and RAM. Using the `-p` flag turns on the pool field; the `-v`(verbose) flag turns on the OS, loader, tap, and con fields; the `-s` (status) flag turns on the VMM, running, and rcboot fields; the `-d` flag turns on the description field; and using the `-l` flag pipes the output to `less`.

To use some of the flags for `info` the command would be `iohyve info -psl` or any order on the flags. The order of the flags does not influence the order of the columns, sorry. Any combination of flags can be used, I wrote the code to be flexible enough for that.

Let me know if you would like to see any further changes. I hope you like the results.

Also @pr1ntf thank you for amusing my curiosity with the number of guests. It was pure curiosity, I re-read the question and realized it could of come off as aggressive. My sincerest apologies if it came off that way, not my intention at all. Also thank you with the follow ups yesterday, it was quiet a long day with little sleep.

## Here is the before and after for `iohyve list`

**Before:**

```
root@bhost:~ # iohyve list
Guest      VMM?  Running  rcboot?  Description
testing    YES   YES      NO       Thu Apr  7 10:37:59 MDT 2016
guest      YES   YES      NO       Ubiquiti mFi Controller Appliance
gst01      NO    NO       NO       Internal https
obsdguest  NO    NO       NO       Wed Apr  6 14:18:44 MDT 2016
```

**After:**

```
root@bhost:~ # iohyve list
Guest      VMM?  Running  rcboot?
testing    YES   YES      NO
guest      YES   YES      NO
gst01      NO    NO       NO
obsdguest  NO    NO       NO
root@bhost:~ # iohyve list -d
Guest      VMM?  Running  rcboot?  Description
testing    YES   YES      NO       Thu Apr  7 10:37:59 MDT 2016
guest      YES   YES      NO       Ubiquiti mFi Controller Appliance
gst01      NO    NO       NO       Internal https
obsdguest  NO    NO       NO       Wed Apr  6 14:18:44 MDT 2016
root@bhost:~ # iohyve list -dv
Guest      VMM?  Running  rcboot?  Tap      Con     Description
testing    YES   YES      NO       tap7011  nmdm17  Thu Apr  7 10:37:59 MDT 2016
guest      YES   YES      NO       tap7003  nmdm1   Ubiquiti mFi Controller Appliance
gst01      NO    NO       NO       tap7004  nmdm10  Internal https
obsdguest  NO    NO       NO       tap7010  nmdm16  Wed Apr  6 14:18:44 MDT 2016
root@bhost:~ # iohyve list -dvl
root@bhost:~ # 
```

## Here is the before and after for `iohyve info`

**Before:**

```
root@bhost:~ # iohyve info
Name             Size  RAM     CPU  OS         Loader
testing          -     256M    1    default    bhyveload
testing/disk0    10G   256M    1    default    bhyveload
guest            -     4G      2    debian     grub-bhyve
guest/disk0      200G  4G      2    debian     grub-bhyve
gst01            -     512M    1    default    bhyveload
gst01/disk0      40G   512M    1    default    bhyveload
obsdguest        -     256M    1    openbsd59  grub-bhyve
obsdguest/disk0  8G    256M    1    openbsd59  grub-bhyve
root@bhost:~ # iohyve info -d
Name             Size  RAM     CPU  OS         Loader      Description
testing          -     256M    1    default    bhyveload   Thu          Apr  7           10:37:59          MDT      2016
testing/disk0    10G   256M    1    default    bhyveload   Thu          Apr  7           10:37:59          MDT      2016
guest            -     4G      2    debian     grub-bhyve  Ubiquiti     mFi  Controller  Appliance
guest/disk0      200G  4G      2    debian     grub-bhyve  Ubiquiti     mFi  Controller  Appliance
gst01            -     512M    1    default    bhyveload   Internal         https
gst01/disk0      40G   512M    1    default    bhyveload   Internal         https
obsdguest        -     256M    1    openbsd59  grub-bhyve  Wed          Apr  6           14:18:44          MDT      2016
obsdguest/disk0  8G    256M    1    openbsd59  grub-bhyve  Wed          Apr  6           14:18:44          MDT      2016
```

**After:**

```
root@bhost:~ # iohyve info             
Guest            Size  CPU  RAM          
testing          -     1    256M         
testing/disk0    10G   1    256M         
guest            -     2    4G           
guest/disk0      200G  2    4G           
gst01            -     1    512M         
gst01/disk0      40G   1    512M         
obsdguest        -     1    256M         
obsdguest/disk0  8G    1    256M         
root@bhost:~ # iohyve info -p          
Guest            Size  CPU  RAM     Pool 
testing          -     1    256M    apool          
testing/disk0    10G   1    256M    apool          
guest            -     2    4G      zpool          
guest/disk0      200G  2    4G      zpool          
gst01            -     1    512M    zpool          
gst01/disk0      40G   1    512M    zpool          
obsdguest        -     1    256M    zpool          
obsdguest/disk0  8G    1    256M    zpool          
root@bhost:~ # iohyve info -pv         
Guest            Size  CPU  RAM     Pool       OS         Loader      Tap      Con                 
testing          -     1    256M    apool      default    bhyveload   tap7011  nmdm17              
testing/disk0    10G   1    256M    apool      default    bhyveload   tap7011  nmdm17              
guest            -     2    4G      zpool      debian     grub-bhyve  tap7003  nmdm1               
guest/disk0      200G  2    4G      zpool      debian     grub-bhyve  tap7003  nmdm1               
gst01            -     1    512M    zpool      default    bhyveload   tap7004  nmdm10              
gst01/disk0      40G   1    512M    zpool      default    bhyveload   tap7004  nmdm10              
obsdguest        -     1    256M    zpool      openbsd59  grub-bhyve  tap7010  nmdm16              
obsdguest/disk0  8G    1    256M    zpool      openbsd59  grub-bhyve  tap7010  nmdm16              
root@bhost:~ # iohyve info -pvs        
Guest            Size  CPU  RAM     Pool       OS         Loader      Tap      Con     VMM?  Running  rcboot?
testing          -     1    256M    apool      default    bhyveload   tap7011  nmdm17  YES   YES      NO     
testing/disk0    10G   1    256M    apool      default    bhyveload   tap7011  nmdm17  NO    NO       NO     
guest            -     2    4G      zpool      debian     grub-bhyve  tap7003  nmdm1   NO    YES      NO     
guest/disk0      200G  2    4G      zpool      debian     grub-bhyve  tap7003  nmdm1   NO    NO       NO     
gst01            -     1    512M    zpool      default    bhyveload   tap7004  nmdm10  NO    NO       NO     
gst01/disk0      40G   1    512M    zpool      default    bhyveload   tap7004  nmdm10  NO    NO       NO     
obsdguest        -     1    256M    zpool      openbsd59  grub-bhyve  tap7010  nmdm16  NO    NO       NO     
obsdguest/disk0  8G    1    256M    zpool      openbsd59  grub-bhyve  tap7010  nmdm16  NO    NO       NO     
root@bhost:~ # iohyve info -pvsd       
Guest            Size  CPU  RAM     Pool       OS         Loader      Tap      Con     VMM?  Running  rcboot?  Description      
testing          -     1    256M    apool      default    bhyveload   tap7011  nmdm17  YES   YES      NO       Thu Apr  7 10:37:59 MDT 2016                  
testing/disk0    10G   1    256M    apool      default    bhyveload   tap7011  nmdm17  NO    NO       NO       Thu Apr  7 10:37:59 MDT 2016                  
guest            -     2    4G      zpool      debian     grub-bhyve  tap7003  nmdm1   NO    YES      NO       Ubiquiti mFi Controller Appliance            
guest/disk0      200G  2    4G      zpool      debian     grub-bhyve  tap7003  nmdm1   NO    NO       NO       Ubiquiti mFi Controller Appliance            
gst01            -     1    512M    zpool      default    bhyveload   tap7004  nmdm10  NO    NO       NO       Internal https            
gst01/disk0      40G   1    512M    zpool      default    bhyveload   tap7004  nmdm10  NO    NO       NO       Internal https            
obsdguest        -     1    256M    zpool      openbsd59  grub-bhyve  tap7010  nmdm16  NO    NO       NO       Wed Apr  6 14:18:44 MDT 2016                  
obsdguest/disk0  8G    1    256M    zpool      openbsd59  grub-bhyve  tap7010  nmdm16  NO    NO       NO       Wed Apr  6 14:18:44 MDT 2016                  
root@bhost:~ # iohyve info -pvsdl
root@bhost:~ # 
```

EpiJunkie commented on iohyve:
```diff
@@ -258,9 +258,25 @@ __get_zfs_pcidev_conf() {
 	local pools="$(zfs list -H | grep iohyve | cut -d '/' -f 1 | uniq )"
 	(
-        printf "%s^%s^%s^%s^%s\n" "Guest" "VMM?" "Running" "rcboot?" "Description"
+	
```
Using this method, the flags are al-la-carte allowing for flexibility and not using as much code as if you were to create an if statement or sub-sub-function for each desired output.

EpiJunkie commented on iohyve:
```diff
@@ -282,27 +298,144 @@ __list() {
 
 # Display info about all guests.
 __info() {
-	dflag="$2"
```
I really do like this bit of code, seriously, so concise with lots of formatted output.

EpiJunkie commented on iohyve:
```diff
@@ -282,27 +298,144 @@ __list() {
             done
 	done
-	) | column -ts^
+	) | column -ts^ | \
```
This was inspired by the function above piping to `column`. I figured the same could be done for a smaller bit of code needing to get piped.

EpiJunkie commented on iohyve:
```diff
@@ -258,9 +258,25 @@ __get_zfs_pcidev_conf() {
+		printf "^%s" "Description"
+	fi
+	
+	printf "\n" # Print new line for first guest.
```
Using this method, the flags can be used al-la-carte without needing to write static blocks of code for the desired output.

EpiJunkie committed:
Fix: Missing line to set $vmm. Added tweak for speed improvement.

Missing line to populate $vmm variable.

Added bypass to not check for vmm, bhyve instance, or boot property
for $name/disk. This causes a slight speed improvment. These values
are irrevevant for disks.

EpiJunkie committed:
Updated new flags available with info & list

Added new available flags under Usage for info and list.

pr1ntf commented:
Great work! Buuuuuuut..... I'm sorry for the confusion, I wanted the `list` function to remain unchanged. 

> Let's keep the list function untouched, but remove description from the info function.
> 
> Maybe info without options will output size, cpu, ram, and pool.
> 
> Info with the -v flag can add os, loader, tap, and con.

I'm so sorry to be the benevolent dictator of iohyve, especially when you've been helping out so much. 
If you've got other ideas and suggestions, I'm all ears.

EpiJunkie commented:
Is it alright if the other flags remain? 

If I am understanding correctly, below is the desired output for flagless and using `-v` flags. The `-l` is there as a request to pipe to `less`. The other flags are there because they make my heart go pitter-pat. In addition, if new properties are added with this code base, they will be easy to flag in without changing the look from previous versions.

### Here is the before and after for `iohyve list` (Revised)

**Before:**

```
root@bhost:~ # iohyve list
Guest      VMM?  Running  rcboot?  Description
testing    YES   YES      NO       Thu Apr  7 10:37:59 MDT 2016
guest      YES   YES      NO       Ubiquiti mFi Controller Appliance
gst01      NO    NO       NO       Internal https
obsdguest  NO    NO       NO       Wed Apr  6 14:18:44 MDT 2016
```

**After:**

```
root@bhost:~ # iohyve list
Guest      VMM?  Running  rcboot?  Description
testing    YES   YES      NO       Thu Apr  7 10:37:59 MDT 2016
guest      YES   YES      NO       Ubiquiti mFi Controller Appliance
gst01      NO    NO       NO       Internal https
obsdguest  NO    NO       NO       Wed Apr  6 14:18:44 MDT 2016
root@bhost:~ # iohyve list -v
Guest      VMM?  Running  rcboot?  Tap      Con     Description
testing    YES   YES      NO       tap7011  nmdm17  Thu Apr  7 10:37:59 MDT 2016
guest      YES   YES      NO       tap7003  nmdm1   Ubiquiti mFi Controller Appliance
gst01      NO    NO       NO       tap7004  nmdm10  Internal https
obsdguest  NO    NO       NO       tap7010  nmdm16  Wed Apr  6 14:18:44 MDT 2016
root@bhost:~ # iohyve list -vl
root@bhost:~ # 
```

### Here is the before and after for `iohyve info` (Revised)

**Before:**

```
root@bhost:~ # iohyve info
Name             Size  RAM     CPU  OS         Loader
testing          -     256M    1    default    bhyveload
testing/disk0    10G   256M    1    default    bhyveload
guest            -     4G      2    debian     grub-bhyve
guest/disk0      200G  4G      2    debian     grub-bhyve
gst01            -     512M    1    default    bhyveload
gst01/disk0      40G   512M    1    default    bhyveload
obsdguest        -     256M    1    openbsd59  grub-bhyve
obsdguest/disk0  8G    256M    1    openbsd59  grub-bhyve
root@bhost:~ # iohyve info -d
Name             Size  RAM     CPU  OS         Loader      Description
testing          -     256M    1    default    bhyveload   Thu          Apr  7           10:37:59          MDT      2016
testing/disk0    10G   256M    1    default    bhyveload   Thu          Apr  7           10:37:59          MDT      2016
guest            -     4G      2    debian     grub-bhyve  Ubiquiti     mFi  Controller  Appliance
guest/disk0      200G  4G      2    debian     grub-bhyve  Ubiquiti     mFi  Controller  Appliance
gst01            -     512M    1    default    bhyveload   Internal         https
gst01/disk0      40G   512M    1    default    bhyveload   Internal         https
obsdguest        -     256M    1    openbsd59  grub-bhyve  Wed          Apr  6           14:18:44          MDT      2016
obsdguest/disk0  8G    256M    1    openbsd59  grub-bhyve  Wed          Apr  6           14:18:44          MDT      2016
```

**After:**

```
root@bhost:~ # iohyve info                
Guest            Size  CPU  RAM     Pool 
testing          -     1    256M    apool          
testing/disk0    10G   1    256M    apool          
guest            -     2    4G      zpool          
guest/disk0      200G  2    4G      zpool          
gst01            -     1    512M    zpool          
gst01/disk0      40G   1    512M    zpool          
obsdguest        -     1    256M    zpool          
obsdguest/disk0  8G    1    256M    zpool          
root@bhost:~ # iohyve info -v         
Guest            Size  CPU  RAM     Pool       OS         Loader      Tap      Con                 
testing          -     1    256M    apool      default    bhyveload   tap7011  nmdm17              
testing/disk0    10G   1    256M    apool      default    bhyveload   tap7011  nmdm17              
guest            -     2    4G      zpool      debian     grub-bhyve  tap7003  nmdm1               
guest/disk0      200G  2    4G      zpool      debian     grub-bhyve  tap7003  nmdm1               
gst01            -     1    512M    zpool      default    bhyveload   tap7004  nmdm10              
gst01/disk0      40G   1    512M    zpool      default    bhyveload   tap7004  nmdm10              
obsdguest        -     1    256M    zpool      openbsd59  grub-bhyve  tap7010  nmdm16              
obsdguest/disk0  8G    1    256M    zpool      openbsd59  grub-bhyve  tap7010  nmdm16              
root@bhost:~ # iohyve info -vs        
Guest            Size  CPU  RAM     Pool       OS         Loader      Tap      Con     VMM?  Running  rcboot?
testing          -     1    256M    apool      default    bhyveload   tap7011  nmdm17  YES   YES      NO     
testing/disk0    10G   1    256M    apool      default    bhyveload   tap7011  nmdm17  NO    NO       NO     
guest            -     2    4G      zpool      debian     grub-bhyve  tap7003  nmdm1   NO    YES      NO     
guest/disk0      200G  2    4G      zpool      debian     grub-bhyve  tap7003  nmdm1   NO    NO       NO     
gst01            -     1    512M    zpool      default    bhyveload   tap7004  nmdm10  NO    NO       NO     
gst01/disk0      40G   1    512M    zpool      default    bhyveload   tap7004  nmdm10  NO    NO       NO     
obsdguest        -     1    256M    zpool      openbsd59  grub-bhyve  tap7010  nmdm16  NO    NO       NO     
obsdguest/disk0  8G    1    256M    zpool      openbsd59  grub-bhyve  tap7010  nmdm16  NO    NO       NO     
root@bhost:~ # iohyve info -vsd       
Guest            Size  CPU  RAM     Pool       OS         Loader      Tap      Con     VMM?  Running  rcboot?  Description      
testing          -     1    256M    apool      default    bhyveload   tap7011  nmdm17  YES   YES      NO       Thu Apr  7 10:37:59 MDT 2016                  
testing/disk0    10G   1    256M    apool      default    bhyveload   tap7011  nmdm17  NO    NO       NO       Thu Apr  7 10:37:59 MDT 2016                  
guest            -     2    4G      zpool      debian     grub-bhyve  tap7003  nmdm1   NO    YES      NO       Ubiquiti mFi Controller Appliance            
guest/disk0      200G  2    4G      zpool      debian     grub-bhyve  tap7003  nmdm1   NO    NO       NO       Ubiquiti mFi Controller Appliance            
gst01            -     1    512M    zpool      default    bhyveload   tap7004  nmdm10  NO    NO       NO       Internal https            
gst01/disk0      40G   1    512M    zpool      default    bhyveload   tap7004  nmdm10  NO    NO       NO       Internal https            
obsdguest        -     1    256M    zpool      openbsd59  grub-bhyve  tap7010  nmdm16  NO    NO       NO       Wed Apr  6 14:18:44 MDT 2016                  
obsdguest/disk0  8G    1    256M    zpool      openbsd59  grub-bhyve  tap7010  nmdm16  NO    NO       NO       Wed Apr  6 14:18:44 MDT 2016                  
root@bhost:~ # iohyve info -vsdl
root@bhost:~ # 
```

pr1ntf commented:
Don't even worry about touching the `description` function at all. 

The info changes look great, though! :+1: 

Again, sorry for the confusion. 

I've done quite a bit to the script the past week, there are conflicts with this PR.

EpiJunkie committed:
Merge with upstream version 0.7.5

EpiJunkie committed:
Fix for eval -l flag for list()

EpiJunkie committed:
Changed default to display pool per pr1ntf in info()

EpiJunkie commented:
I am guessing you meant `list` function. I left the `-l` flag to pipe to less. Let me know if that needs to be removed too.

Thanks.

pr1ntf commented:
Lookin good in the neighborhood.
